pkg/csi/service/fcd/controller.go

```diff
@@ -72,15 +74,112 @@ func (c *controller) CreateVolume(
 	req *csi.CreateVolumeRequest) (
 	*csi.CreateVolumeResponse, error) {
 
-	return nil, nil
+	params := req.GetParameters()
+
+	// Volume Name
+	volName := req.GetName()
+	if len(volName) == 0 {
+		myUUID, err := uuid.NewUUID()
+		if err != nil {
+			log.Errorf("Failed to generate a UUID. Err: %v", err)
+			return nil, err
+		}
+		volName = myUUID.String()
+	}
+
+	// Volume Size - Default is 10 GiB
+	volSizeBytes := int64(DefaultGbDiskSize * GbInBytes)
+	if req.GetCapacityRange() != nil {
+		volSizeBytes = int64(req.GetCapacityRange().GetRequiredBytes())
+	}
```

Comment:
So this gets slightly complicated. I'd have to dig into the generated protobuf code, but I think `req.GetCapacityRange()` can return non-nil, but still have zero values for required and limit fields.

So within this `if` statement, we need to see if `GetRequiredBytes` returns 0. If it does, use the default size. And *then* we also need to see if the the capacity limit has been set, and ensure we are not over the limit.  I don't think the spec validator would catch this...

For example, a request could set the required bytes to 0GB, but set a limit of 8GiB, and that would be an invalid request for us with that default value. what do we do then? Use 8GiB? Or return an error? I'd wager the former.

Reply:
Will make the change.